src/uiprotect/api.py

```diff
@@ -1538,6 +1666,64 @@ async def callback(total: int, chunk: bytes | None) -> None:
         r.close()
         return None
 
+    async def get_camera_video(
+        self,
+        camera_id: str,
+        start: datetime,
+        end: datetime,
+        channel_index: int = 0,
+        validate_channel_id: bool = True,
+        output_file: Optional[Path] = None,
+        iterator_callback: Optional[IteratorCallback] = None,
+        progress_callback: Optional[ProgressCallback] = None,
+        chunk_size: int = 65536,
+        fps: Optional[int] = None,
+        filename: Optional[str] = None,
+    ) -> Optional[bytes]:
+        """
+        Deprecated: maintained for backwards compatibility.
+
+        If you are using Unifi Protect 4 or later, please use
+        prepare_camera_video() and download_camera_video() instead.
+        """
+        try:
+            prepare_response = await self.prepare_camera_video(
+                camera_id=camera_id,
+                start=start,
+                end=end,
+                channel_index=channel_index,
+                validate_channel_id=validate_channel_id,
+                fps=fps,
+                filename=filename,
+            )
+
+            if isinstance(prepare_response, dict):
+                download_filename = prepare_response["fileName"]
+            else:
+                raise Exception
+
+            return await self.download_camera_video(
+                camera_id=camera_id,
+                filename=download_filename,
+                output_file=output_file,
+                iterator_callback=iterator_callback,
+                progress_callback=progress_callback,
+                chunk_size=chunk_size,
+            )
+        except Exception:
```

Comment:
Please avoid a broad exception catch and only catch exceptions that would confirm the need to fallback to the other method as it can unexpectedly hide bugs

Reply:
This was quite intentional, as it's supposed to fall back on the previous video/export on any kind of error, including using it on a pre-4 Protect installation whose responses I can't predict since I have no platform to test it on.  The get_camera_video() needs to be deprecated in favor of either using the export_camera_video method, which is functionally identical, or the new prepare/download_camera_video methods used in Protect 4.

Reply:
It's better for it to fail than to have undefined behavior since if it falls back without warning the problem will never get fixed, and when the deprecated function is removed we will find out the hard way when it breaks which will be harder to fix.

Please narrow the exception catch to only exceptions that are expected. 

https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/broad-exception-caught.html

Reply:
This also needs a test to simulate a failure so we know under which conditions it should fallback to the old api

Reply:
It should also log a message when it falls back so we know it's happening so we have a chance to fix whatever is causing the failure